src/stan/lang/ast/fun/has_non_param_var_vis_def.hpp

```diff
@@ -93,6 +93,17 @@ namespace stan {
         || boost::apply_visitor(*this, e.theta_.expr_);
     }
 
+    bool has_non_param_var_vis::operator()(const algebra_solver& e) const {
+      // if any vars, return true  -- CHECK: nonlinearity?
```

Comment:
it does. y_ is theta, sorry about the confusing notation. These are the terms we used in the grant proposal, when explaining the implicit function theorem:
x -- unknown
y -- auxiliary parameters
z -- returned value by the algebraic function